src/eon_type_error.erl

```diff
@@ -57,30 +57,26 @@ get_full_key_test() ->
         , [ { <<"contact_info">>, type_contact_info, contact_info, the_term
             , { untypable
               , [{<<"email">>, type_email, email, the_term, validate}]}}]},
-  [<<"contact_info">>, <<"email">>] = get_error_key_list(Rsn),
-  <<"contact_info.email">>         = get_full_key(Rsn).
+  ?assertEqual([<<"contact_info">>, <<"email">>], get_error_key_list(Rsn)),
+  ?assertEqual(<<"contact_info.email">>, get_full_key(Rsn)).
 
 get_root_cause__primitive_type__test() ->
-  {error, Rsn} = eon_type:check_term(42, type_string),
-  {42, type_string} = get_root_cause(Rsn).
+  {error, Rsn} = eon_type:check_term(42, test_string),
+  ?assertEqual({42, test_string}, get_root_cause(Rsn)).
 
 get_root_cause__recursive_type_wrong_nested_type__test() ->
-  Term =
-    [ {<<"orgnr">>, <<"SE1234">>}
-    , {<<"name">>, <<"Test">>}
-    ],
-  {error, Rsn} = eon_type:check_term(Term, type_company_id_item),
-  {<<"SE1234">>, type_orgnr} = get_root_cause(Rsn).
+  Address = [ {<<"street">>, <<"Main Street">>}
+            , {<<"residents">>, [42, <<"Amelie">>, <<"Karin">>]}],
+  {error, Rsn} = eon_type:check_term(Address, test_address),
+  ?assertEqual({42, test_string}, get_root_cause(Rsn)).
```

Comment:
```suggestion
  Address = [ {<<"street">>, 42}
            , {<<"residents">>, [<<"Amelie">>, <<"Karin">>]}],
  {error, Rsn} = eon_type:check_term(Address, test_address),
  ?assertEqual({42, test_string}, get_root_cause(Rsn)).
```

This would have been an equal replacement for testing a recursive type without diving into a list. A test for lists is already below as you know.